src/IO/Resources/FontsLoader.cs

```diff
@@ -543,66 +546,71 @@ ushort flags
 
             ref FontCharacterData[] fd = ref _fontData[font];
 
-            StringBuilder sb = new StringBuilder();
+            int strLen = str.Length;
 
-            if (IsUsingHTML)
+            ValueStringBuilder sb = new ValueStringBuilder(strLen);
```

Comment:
If we can predetermine the end size of the string, we can avoid resizes, which means we generally don't need rented buffers unless the Span<char> will be large.
There is a break-even point where stackallocating a buffer is slower than calling out to the array pool. This is because for right now, stack allocating always clears the the buffer. IIRC, it's like 100KB or so which is usually plenty.
So if needed, you can do an explicit check for the size of the final string (if it is known), and either stackalloc, or rent accordingly.

Reply:
makes sense. I'll check it out